### PR TITLE
feat(NativeSelect): Updated to match other form components

### LINKED ISSUE #919: NativeSelect: Refactor to use new form styling and elements
We are deprecating `InputWrapper` due to future support for css library.

Use the same elements and styling that is present in new `Textfield` for `NativeSelect`.

Apart from prefix/suffix, it should have a very similar component api as `Textfield` / `Textarea`.

### PR DESCRIPTION
**Pull request by Barsnes**
Resolves #919

Removed use of `InputWrapper`, as this component will be deprecated.
It now works with `Fieldset`, and matches other form components

### CODE REVIEW

**Comment by Barsnes** [comment]
I believe I have implemented your wishes @mimarz. Let me know if anything is missing/wrong

File: packages/react/src/components/form/NativeSelect/NativeSelect.module.css [hunk 1]
```diff
@@ -0,0 +1,64 @@
+.input {
```
Comment (mimarz) [inline packages/react/src/components/form/NativeSelect/NativeSelect.module.css:1]: Suggest renaming this class to `select`

File: packages/react/src/components/form/NativeSelect/NativeSelect.module.css [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  font: inherit;
+  position: relative;
+  box-sizing: border-box;
+  flex: 0 1 auto;
+  min-height: 2.5em;
+  width: 100%;
+  padding: 0 var(--fds-spacing-3);
```
Comment (mimarz) [inline packages/react/src/components/form/NativeSelect/NativeSelect.module.css:8]: Looked like to much padding with `spacing-3`, suggest `spacing-2`
```suggestion
  padding: 0 var(--fds-spacing-2);
```

File: packages/react/src/components/form/NativeSelect/NativeSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+   * @default false
+   * */
+  hideLabel?: boolean;
+  /** Set to true to enable multiple selection. */
+  multiple?: boolean;
+  /** The ID of the `select` element. This will be generated if not provided. */
+  id?: string;
+  /**
+   * Defines the size of the select.
+   * @default 'medium'
```
Comment (mimarz) [inline packages/react/src/components/form/NativeSelect/NativeSelect.tsx:31]: Remove the ticks as its showing wrong value in props table.
image

```suggestion
   * @default medium
```

File: packages/react/src/components/form/NativeSelect/NativeSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import classes from './NativeSelect.module.css';
+import utilityClasses from './../../../utils/utility.module.css';
+import { useNativeSelect } from './useNativeSelect';
+
+export type NativeSelectProps = {
+  /**
+   * Specifies whether the select box is disabled.
+   * @default false
+   * */
+  disabled?: boolean;
```
Comment (mimarz) [inline packages/react/src/components/form/NativeSelect/NativeSelect.tsx:18]: Suggest removing this to be consistent with other components, as we haven't documented other untouched native props

File: packages/react/src/components/form/NativeSelect/NativeSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        className={cn(
+          classes.formField,
+          disabled && classes.disabled,
+          readOnly && classes.readOnly,
+          error && classes.error,
+        )}
+      >
+        {label && (
+          <Label
+            weight='medium'
```
Comment (mimarz) [inline packages/react/src/components/form/NativeSelect/NativeSelect.tsx:74]: missing `size={size}` prop

**Review by mimarz — CHANGES_REQUESTED** [review]

**Review by mimarz — APPROVED** [review]
power-power-rangers

**Pull request merged**